[#63686] update health status report documentation (#18907)
opf/openproject: docs/system-admin-guide/files/external-file-storages/health-status/README.md

Author: 💡 I moved the common text above nc and od sections. There was a lot of duplicate description. In general, only the error codes differ, the rest is identical for NC and OD storages. (And we even have identical error codes, which do have a separate section).

# Health status checks and troubleshooting

If a file storage is not working as expected, you can find additional information about possible errors in the details
view of the file storage. You can access this view by clicking on the file storage's name in the list under
*Administration* → *Files* → *External file storages*. There, administrators can manually trigger a connection
validation and receive a health status report, or review any errors reported by background synchronization workers
needed for file storages with automatically managed project folders.

## Connection validation

> Every file storage for OneDrive/SharePoint is able to run connection checks. This test is triggered manually by

Reviewer: ```suggestion
Every file storage for OneDrive/SharePoint and Nextcloud is able to run connection checks. This test is triggered manually by
```

clicking on **Run checks now** in the sidebar on the right side of the file storage's details view. This check is
available from the UI to edit the storage.

![Check connection for a file storage in OpenProject administration](openproject_file_storages_onedrive_run_checks_button.png)

> Once the check is finished, a full health report will be generated and a brief summary is displayed. Click **Open full

Reviewer: ```suggestion
Once the check is finished, a full health report will be generated and a brief summary will be displayed. Click **Open full
```

| Error code         | Error description                         | Possible reasons                                                            | Next steps and solutions                                                                                                                 |
|--------------------|-------------------------------------------|-----------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------|
| ERR_NOT_CONFIGURED | The file storage is not fully configured. | Important data is missing, so that the file storage is labelled incomplete. | Check the input fields and fill in the missing data.                                                                                     |
| ERR_UNKNOWN_ERROR  | An unknown error occurred.                | There can be multiple reasons and the error source was not foreseen.        | Errors of this kind are logged to the server logs. Look for a log entry starting with `Connection validation failed with unknown error:` |

### Error codes specific for OneDrive/SharePoint

> The following table lists the error codes of failures specific for OneDrive/SharePoint, with a description of the

Reviewer: ```suggestion
The following table lists the error codes of failures specific to OneDrive/SharePoint, with a description of the
```

| WRN_OD_UNEXPECTED_CONTENT              | The connection request was successful, but unexpected content was found in the drive. | This warning is only shown, if the file storage is configured to automatically managed project folder permissions. There was data found in the drive, that is not a project folder created by OpenProject. | Go to your drive and migrate or delete the data from the drive root, that was not created by OpenProject. Further information about the unexpected data is found in the server logs. A drive configured for usage with the *Automatically managed project folders* option has a disrupted inheritance chain. Any data in here can only be seen by site owner. It is discouraged to use this drive for other purposes than the OpenProject integration. |

### Error codes specific for Nextcloud

> The following table lists the error codes of failures specific for Nextcloud, with a description of the possible reasons

Reviewer: ```suggestion
The following table lists the error codes of failures specific to Nextcloud, with a description of the possible reasons
```

| WRN_NC_UNEXPECTED_CONTENT          | The connection request was successful, but unexpected content was found in the group folder. | The group folder `OpenProject` might contain data, that was put there by a user, or there are remnants from projects that no longer have a valid connection in OpenProject. | Go to Nextcloud and migrate or delete the data in the OpenProject group folder, that was not created by OpenProject. Further information about the unexpected data is found in the server logs.             |

The officially minimal supported app versions are listed in
the [system admin guide](../../../../system-admin-guide/integrations/nextcloud/#required-system-versions).

### Error codes specific for file storages with SSO configuration

> Some file storages can be configured to use SSO login methods. There are a couple of checks run explicitly for this kind

Reviewer: ```suggestion
Some file storages can be configured to use SSO login methods. There are a few checks run specifically for this type of
```

> Some file storages can be configured to use SSO login methods. There are a couple of checks run explicitly for this kind

Reviewer: ```suggestion
Some file storages can be configured to use SSO login methods. There are a few checks run specifically for this type of
```

> of setup. The following table contains the error codes of those checks.

Reviewer: ```suggestion
of setup. The following table lists the error codes related to those checks.
```

clicking on **Run checks now** in the sidebar on the right side of the file storage's details view. This check is
available from the UI to edit the storage.

![Check connection for a file storage in OpenProject administration](openproject_file_storages_onedrive_run_checks_button.png)

Once the check is finished, a full health report will be generated and a brief summary is displayed. Click **Open full
health report** to see the report in full detail and to download it.

![Link to open full health report for a file storage in OpenProject administration](openproject_file_storages_onedrive_open_full_health_report.png)

The full health status report will give an overview of all checks that were performed grouped into three categories:

- Basic configuration
- Authentication
- Automatically managed project folder configuration (if activated for this file storage)

In the top right corner you can **Re-run all checks** or **Download** the health report in a text format by clicking the
respective buttons.

![Health status report for a file storage in OpenProject administration](openproject_file_storages_onedrive_open_full_health_report_download_button.png)

> > [!TIP]

Reviewer 2: @MayaBerd @Kharonus I added a tip section
